mail_multicompany/models/mail_message.py

```diff
@@ -19,14 +19,20 @@ def create(self, vals_list):
             if not vals.get("company_id"):
                 vals["company_id"] = self.env.company.id
             if not vals.get("mail_server_id"):
-                vals["mail_server_id"] = (
-                    self.sudo()
-                    .env["ir.mail_server"]
-                    .search(
-                        [("company_id", "=", vals.get("company_id", False))],
-                        order="sequence",
-                        limit=1,
-                    )
-                    .id
+                mail_server, email_from = self._find_company_mail_server(
+                    vals.get("email_from", False), vals.get("company_id", False)
                 )
+                vals["mail_server_id"] = mail_server.id if mail_server else False
+                vals["email_from"] = email_from
         return super(MailMessage, self).create(vals_list)
+
+    def _find_company_mail_server(self, email_from, company_id):
+        # We want to find the most appropriate mail server for the given email_from
+        # but limiting it to the current active company mail servers.
+        mail_servers = (
+            self.sudo().env["ir.mail_server"].search([("company_id", "=", company_id)])
+        )
+        if not mail_servers:
+            return False, email_from
```

Comment:
```suggestion
            return None, email_from
```
as the same odoo does in _find_mail_server